### PR TITLE
update for checkErrorModal method

### LINKED ISSUE #1179: use XCUIElementTypeAlert for appium tests of modals
to this method
Screenshot 2021-12-06 at 13 37 16
but i dont know how to add it

_Originally posted by @fcvakintos in https://github.com/FlowCrypt/flowcrypt-ios/issues/1162#issuecomment-986694763_

### PR DESCRIPTION
**Pull request by fcvakintos**
This PR contains changes for checkErrorModal method, updated mock and live tests
close #1179 

----------------------------------

**Tests** 
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: appium/tests/screenobjects/base.screen.ts [hunk 1]
```diff
@@ -6,7 +6,8 @@ const SELECTORS = {
   ERROR_HEADER: '-ios class chain:**/XCUIElementTypeStaticText[`label == "Error"`]',
   OK_BUTTON: '~OK',
   ERROR_FES_HEADER: '-ios class chain:**/XCUIElementTypeStaticText[`label == "Startup Error"`]',
-  RETRY_BUTTON: '~Retry'
+  RETRY_BUTTON: '~Retry',
+  CURRENT_ERROR: '-ios predicate string:type == "XCUIElementTypeAlert"'
```
Comment (sosnovsky) [inline appium/tests/screenobjects/base.screen.ts:10]: It'll be better to call this selector `CURRENT_MODAL`, as in some cases there won't be errors, just some app message.

File: appium/tests/screenobjects/base.screen.ts [hunk 2]
```diff
@@ -40,19 +33,9 @@ export default class BaseScreen {
...
-    await expect(await this.okButton).toBeDisplayed();
-  }
-
-  static checkErrorModalForFES = async (errorText: string) => {
-    const message = '-ios class chain:**/XCUIElementTypeAlert/XCUIElementTypeOther/XCUIElementTypeOther/' +
-        'XCUIElementTypeOther[2]/XCUIElementTypeScrollView[1]/XCUIElementTypeOther[1]/XCUIElementTypeStaticText[2]';//it works only with this selector
-    await expect(await this.errorFESHeader).toBeDisplayed();
-    expect(await (await $(message)).getAttribute('value')).toContain(errorText);
-    await expect(await this.retryButton).toBeDisplayed();
+    await expect(await this.currentModal).toBeDisplayed();
```
Comment (sosnovsky) [inline appium/tests/screenobjects/base.screen.ts:36]: Here better naming will be `checkModalMessage` with `message` parameter, as we'll check not only errors, but other messages too.

**Review by sosnovsky — CHANGES_REQUESTED** [review]

File: appium/tests/screenobjects/base.screen.ts (hunk 2 above)
Comment (fcvakintos) [reply appium/tests/screenobjects/base.screen.ts:36]: ok, updated

File: appium/tests/screenobjects/base.screen.ts (hunk 1 above)
Comment (fcvakintos) [reply appium/tests/screenobjects/base.screen.ts:10]: updated

**Review by sosnovsky — APPROVED** [review]

**Review by sosnovsky — APPROVED** [review]

**Pull request merged**